### PR TITLE
Model-Agent interaction API improvement / docs

### PR DESCRIPTION
**Pull request by Datseris**
Improves the API and documentation a lot. It is crazy how much cleaner the source code becomes when one realizes which functions should be calling which. Have a look at the source around `add_agent!` for continuous and discrete space and the continuous case is so much cleaner.

### CODE REVIEW

**Comment by Datseris** [comment]
@kavir1698 is this good to or you haven't reviewed yet?

**Comment by kavir1698** [comment]
I had written the comments but had not actually submitted them.

File: src/core/continuous_space.jl [hunk 1]
```diff
@@ -95,7 +98,7 @@ function collect_ids(q::SQLite.Query, colname=:id)
   return output
 end
 
-# TODO: index! needs a better name
+# TODO: index! needs a better name, and consider whether this should be part of public API
```
[the remark this answers is not in the document]
Comment (Datseris) [reply src/core/continuous_space.jl]: alright then we should add a docstring and add it to the ContinuousSPace API

File: src/core/agent_space_interaction.jl [hunk 2]
```diff
@@ -23,16 +29,44 @@ function kill_agent!(agent::A, model::ABM{A, Nothing}) where A
...
 """
-    move_agent!(agent::AbstractAgent [, pos], model::ABM) → agent
+genocide!(model::ABM) = for (i, a) in model.agents; kill_agent!(a, model); end
 
-Add `agentID` to the new position `pos` (either tuple or integer) in the model
+"""
+    genocide!(model::ABM, n::Int)
+Kill the agents of the model whose IDs are larger than n.
+"""
+function genocide!(model::ABM, n::Int)
```
[the remark this answers is not in the document]
Comment (Datseris) [reply src/core/agent_space_interaction.jl]: Sorry, where are you referring to? I can't seem to find it. Can you restore it?

[the remark this answers is not in the document]
Comment (kavir1698) [reply src/core/agent_space_interaction.jl]: Actually, it seems ok. `genocide!` functions in `core/continuous_space.jl` are the same.

**Pull request merged**